UOP1_Project/Assets/Scripts/Characters/WalkingParticle.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class WalkingParticle : MonoBehaviour
+{
+
+    [Tooltip("Particle prefab to be attached to the charactor in the scene.")] public ParticleSystem walkingParticlePrefab;
+
+    private Vector2 lastPosition;
+    private Vector2 currentPosition;
+    private ParticleSystem particleObject;
+
+    private void OnEnable()
+    {
+        lastPosition = new Vector2(transform.position.x, transform.position.z);
+        currentPosition = new Vector2(transform.position.x, transform.position.z);
+    }
+
+    private void Update()
+    {
+        if (!particleObject && walkingParticlePrefab)
```

Comment:
Why not do this in OnEnable too? You could also destroy the instantiated system in OnDisable, so there's a straightforward symmetry.

Reply:
(If you do it in OnEnable then I think you will be able to just assume that it is not null in Update, so you won't need the if-check for it at all any more).